crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1399,6 +1406,20 @@ impl CumulocityConverter {
         &mut self,
         message: &MqttMessage,
     ) -> Result<Vec<MqttMessage>, ConversionError> {
+        if self.active_commands_last_cleared.elapsed() > Duration::from_secs(3600) {
+            let mut to_remove = vec![];
+            for (id, time) in &self.active_commands {
+                // Expire tasks after 12 hours
+                if time.elapsed() > Duration::from_secs(3600 * 12) {
+                    to_remove.push(id.to_owned());
```

Comment:
Although it's highly unlikely for a normal operation to keep running for over 12 hours, there could be cases where an operation like firmware update may get delayed by the workflow because it is scheduled to be executed during non-busy hours(late night/early morning). For e.g, the device receives the firmware update at 9 AM in the morning, but postpones the execution because the device is "busy" until 11 PM at night, or until some point in future where it can decide that the device is "not busy". In such cases, this 12 hours assumption might not be safe enough.

An easier solution would be to use an even higher value, but even that would be arbit. The safest option would be not to remove until the the operation has fully completed (in its terminal state or cleared)

Reply:
This cache only prevents creation of multiple commands for an operation and stores no state. So clearing the cache will no break a pending operation.

Reply:
We do also remove operations when they are complete, but to do this for c8y custom operations too we would need to encode the logic for how c8y processes incoming smartrest commands (including updating the operation by either type or id). Because we only use this for de-duplication, it really only need represent pending operations, non-completed operations is overkill.